File: javascript/kindergarten-garden/kindergarten-garden.js
const Garden = function(layout, students = Garden.students) {
  const garden = layout.split('\n').map((row) => {
    return [...row].map((plant) => Garden.plants[plant]);
  });
  students = students.sort();
  let out = {};
  students.forEach((student, index) => {
    let gi = index * 2;
    plants = [garden[0][gi], garden[0][gi+1], garden[1][gi], garden[1][gi+1]]
    out[student.toLowerCase()] = plants;
  });
  return out;
};

Garden.plants = {
  G: 'grass',
  V: 'violets',
  R: 'radishes',
  C: 'clover'
};

Garden.students = [
  'Alice',
  'Bob',
  'Charlie',
  'David',
  'Eve',
  'Fred',
  'Ginny',
  'Harriet',
  'Ileana',
  'Joseph',
  'Kincaid',
  'Larry'
];

module.exports = Garden;

File: python/kindergarten-garden/kindergarten_garden.py
PLANTS   = { 'C': 'Clover', 'G': 'Grass', 'R': 'Radishes', 'V': 'Violets' }
STUDENTS = [ 'Alice', 'Bob', 'Charlie', 'David', 'Eve', 'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph', 'Kincaid', 'Larry' ]

class Garden(object):
    def __init__(self, garden, students=STUDENTS):
        garden = [[PLANTS[q] for q in p] for p in garden.split()]
        garden_length = len(garden[0]) // 2

        students.sort()
        self.student_garden = {}
        for i, student in enumerate(students[:garden_length]):
            garden_index = i*2
            self.student_garden[student] = [
                garden[0][garden_index],
                garden[0][garden_index+1],
                garden[1][garden_index],
                garden[1][garden_index+1]
            ]

    def plants(self, student):
        return self.student_garden[student]
